docs/InstrumentationVerification/InstrumentationVerification.md

```diff
@@ -0,0 +1,134 @@
+# Instrumentation Verification Library
+
+
+# Description
+
+The Instrumentation Verification library allows us to generate an assembly on disk which contains all the metadata and bytecode changes that were performed at runtime through the CLR Profiling API, and then run that assembly through various tools that verify the correctness of the instrumented assembly.
+
+
+The library is meant to be used in several distinct two-cases:
+
+
+
+1. As part of CI, the tool can be used:
+    1. By the Tracer team, to verify that we produce valid IL for every integration we add
+    2. By the Debugger team, to verify that adding probes produces valid IL on a wide range of methods
+2. As a diagnostic tool, the InstrumentationVerification tool can be used to quickly triage support cases where the customer is complaining that our CLR Profiler causes a `BadImageFormationException`, `InvalidProgramException`, `TypeLoadException`, etc etc. 
+
+    A primary goal of this tool is to support the scenario where we need to perform a post-mortem diagnosis of a crash that we suspect may have been caused due to faulty instrumentation (for example, a crash due to an ExecutionEngineException or an AccessViolationException). 
+
+3. As part of development of new bytecode instrumentation techniques, enabling breakpoint debugging of instrumented methods via [dnSpy](https://github.com/dnSpy/dnSpy).  
+
+
+# Usage
+
+To run a verification, simply:
+
+
+
+1. Add the environment variable `DD_TRACE_WRITE_INSTRUMENTATION_TO_DISK`, set to “true”.
+2. Run or restart the application we are instrumenting. Observe that as we instrument the app, more and more files will appear under [Datadog Logs Folder]/InstrumentationVerification/{ProcessName}_{ProcessID}_{ProcessCreationTime}
+3. Via command line, run the command: 
+
+    Datadog.InstrumentedAssemblyVerification.Standalone.exe [Full path of the folder generated in step 2] 
+
+
+    Once the command has finished executing, you will find new files under the InstrumentationVerification folder:
+
+    1. Logs file of the verification tool for each assembly will appear under a folder named **OUPUT_InstrumentedAssemblies/InstrumentationVerificationLogger_[DATE]_.log**)
+    2. The instrumented assemblies will appear under a folder named **OUPUT_InstrumentedAssemblies**
+
+**Note:** When handling a support case, the first two steps must be performed on the customer’s site. However, if the customer is willing to upload and send us the instrumentation output folder (which is conveniently placed inside the Datadog Logs folder), then the 3rd step can be performed offline.
+
+
+# Approach & High Level Design
+
+
+![alt_text](InstrumentationVerificationDesign.png "instrumentation verification high level design")
```

Comment:
nit- on a dark background the arrows aren't super visible

Reply:
Fixed in 4c7c44dcf9bc4f5affbde6c20b8fd17d1735d225